homeassistant/components/climate/__init__.py

```diff
@@ -948,6 +990,54 @@ async def async_service_temperature_set(
         else:
             kwargs[value] = temp
 
+    if not hasattr(entity, "min_temperature_range"):
+        await entity.async_set_temperature(**kwargs)
+        return
+
+    if (
+        (min_temp_range := entity.min_temperature_range)
+        and (target_low_temp := kwargs.get(ATTR_TARGET_TEMP_LOW))
+        and (target_high_temp := kwargs.get(ATTR_TARGET_TEMP_HIGH))
+        and (target_high_temp - target_low_temp) < min_temp_range
+    ):
+        # Ensure target_low_temp is not higher than target_high_temp.
+        if target_low_temp > target_high_temp:
+            raise ServiceValidationError(
+                translation_domain=DOMAIN,
+                translation_key="low_temp_higher_than_high_temp",
+            )
+
+        # Ensure deadband between target temperatures.
+        initial_low_temp = target_low_temp
+        if (target_high_temp - target_low_temp) < min_temp_range:
+            target_low_temp = target_high_temp - min_temp_range
+
+        # Guard target_low_temp isn't lower than entity low_temp
+        if target_low_temp < min_temp:
+            raise ServiceValidationError(
+                translation_domain=DOMAIN,
+                translation_key="outside_tolerance",
```

Comment:
Let's use the same language, I think the translation string and the error message should talk about `min_temperature_range`

Reply:
Has been removed